### PR TITLE
EFR-818 Activity receive report, send ack

### PR DESCRIPTION
**Pull request by psjobergprivat**

### CODE REVIEW

File: application/src/main/java/eu/europa/ec/fisheries/uvms/activity/message/consumer/bean/ActivityMessageConsumerBean.java [hunk 1]
```diff
@@ -51,44 +56,60 @@ This file is part of the Integrated Fisheries Data Management (IFDM) Suite. The
...
+
     @EJB
     private FluxReportMessageSaver fluxReportMessageSaver;
 
+    @EJB
+    private ExchangeProducerBean exchangeProducerBean;
+
     @Override
     public void onMessage(Message message) {
         log.debug("Received subscription message");
```
Comment (RobinMilding) [inline application/src/main/java/eu/europa/ec/fisheries/uvms/activity/message/consumer/bean/ActivityMessageConsumerBean.java]: This debug message is misleading

File: application/src/main/java/eu/europa/ec/fisheries/uvms/activity/message/producer/ExchangeProducerBean.java [hunk 2]
```diff
@@ -10,24 +10,41 @@ This file is part of the Integrated Fisheries Data Management (IFDM) Suite. The
...
+
+    public void sendEfrReportSavedAckToExchange(UUID ackResponseMessageId) throws JMSException {
+        if (ackResponseMessageId == null) {
+            return;
+        }
+        EfrReportSaved efrReportSaved = new EfrReportSaved();
+        efrReportSaved.setMethod(ExchangeModuleMethod.EFR_REPORT_SAVED);
+        efrReportSaved.setRefGuid(ackResponseMessageId.toString());
+
+        String xml = JAXBMarshaller.marshallJaxBObjectToString(efrReportSaved);
```
Comment (RobinMilding) [inline application/src/main/java/eu/europa/ec/fisheries/uvms/activity/message/producer/ExchangeProducerBean.java:47]: Don't we want the ACK response to be JSON?

**Review by RobinMilding — CHANGES_REQUESTED** [review]

File: application/src/main/java/eu/europa/ec/fisheries/uvms/activity/message/consumer/bean/ActivityMessageConsumerBean.java (hunk 1 above)
Comment (psjobergprivat) [reply application/src/main/java/eu/europa/ec/fisheries/uvms/activity/message/consumer/bean/ActivityMessageConsumerBean.java]: Definitely, I'll change

File: application/src/main/java/eu/europa/ec/fisheries/uvms/activity/message/producer/ExchangeProducerBean.java (hunk 2 above)
Comment (psjobergprivat) [reply application/src/main/java/eu/europa/ec/fisheries/uvms/activity/message/producer/ExchangeProducerBean.java:47]: The EfrReportSaved-message is defined in an xml schema in exchange-model and I think Exchange expects it to be xml in order to parse and save it

**Review by RobinMilding — APPROVED** [review]

**Pull request merged**